jax/_src/numpy/lax_numpy.py

```diff
@@ -5567,3 +5567,34 @@ def put(a: ArrayLike, ind: ArrayLike, v: ArrayLike,
   else:
     raise ValueError(f"mode should be one of 'wrap' or 'clip'; got {mode=}")
   return arr.at[unravel_index(ind_arr, arr.shape)].set(v_arr, mode=scatter_mode)
+
+
+@util.implements(getattr(np, 'cumulative_sum', None))
+def cumulative_sum(
```

Comment:
Consider defining this function in `numpy/reductions.py`.

Reply:
I defined it here to avoid circular imports due to requiring `jax.numpy.concat`

Reply:
Maybe use `lax.concatenate` instead? `jnp.concat` is just a wrapper around this with some additional validation, which I don't think is needed in this case.